lib/fastlane/actions/recreate_schemes.rb

```diff
@@ -0,0 +1,36 @@
+module Fastlane
+  module Actions
+    class RecreateSchemesAction < Action
+      def self.run(params)
+        require 'xcodeproj'
+
+        Helper.log.info "Recreate schemes for project: #{params[:project]}"
+
+        project = Xcodeproj::Project.open(params[:project])
+        project.recreate_user_schemes
```

Comment:
Nope when manipulating schemes, a `project.save` isn't needed because the `recreate_user_schemes` doesn't modify the `.xcodeproj/project.pbxproj` project file itself, but instead recreate `xcuserdata/xcschemes/*.xcscheme` files on disk; in Xcode, schemes are stored in separate files, not in the Xcode project file itself.